### PR TITLE
[PLDM] Resolved nullptr dereference errors in PLDM

### PR DESCRIPTION
**Pull request by ArchanaKakani**
Resolved nullptr dereference errors.

Signed-off-by: ArchanaKakani <archana.kakani@ibm.com>

### CODE REVIEW

File: libpldm/pdr.c [hunk 1]
```diff
@@ -64,12 +73,10 @@ static void add_hotplug_record(pldm_pdr *repo, pldm_pdr_record *record,
...
-		record->next = curr->next;
-		curr->next = record;
-		if (record->next == NULL) {
-			repo->last->next = record;
-			repo->last = record;
-		}
+	}
+	if (recordAdded == false) {
+		free(record);
+		assert(recordAdded != false);
```
Comment (nmuruli) [inline libpldm/pdr.c:79]: We are going to force assert ie PLDM core here?

File: libpldm/pdr.c [hunk 2]
```diff
@@ -1126,17 +1138,17 @@ void pldm_entity_association_tree_delete_node(
...
+			} else {
+				prev->next_sibling = start->next_sibling;
+			}
+			start->next_sibling = NULL;
 			break;
 		}
 		prev = start;
 		start = start->next_sibling;
 	}
-	if (start == parent->first_child) {
```
Comment (nmuruli) [inline libpldm/pdr.c:1134]: Can you please post what error message we see here and are trying to solve

File: libpldm/pdr.c [hunk 3]
```diff
@@ -1707,6 +1719,12 @@ uint32_t pldm_entity_association_pdr_add_contained_entity(
 			prev = curr;
 			curr = curr->next;
 		}
+		if (curr == NULL) {
+			printf("Couldn't find bmc record handle %d\n",
```
Comment (nmuruli) [inline libpldm/pdr.c:1723]: remove printf?

File: libpldm/pdr.c [hunk 4]
```diff
@@ -1707,6 +1719,12 @@ uint32_t pldm_entity_association_pdr_add_contained_entity(
...
+			       bmc_record_handle);
+			free(new_record);
```
Comment (nmuruli) [inline libpldm/pdr.c:1725]: Can we instead make "added=false" and at line 1774 already free is done based on the "added" flag

File: libpldm/pdr.c (hunk 2 above)
Comment (ArchanaKakani) [reply libpldm/pdr.c:1134]: This is the error msg : 
libpldm/pdr.c:1139:2: warning: Either the condition 'start!=NULL' is redundant or there is possible null pointer dereference: start. [nullPointerRedundantCheck]
 start->next_sibling = NULL;

File: libpldm/pdr.c (hunk 1 above)
Comment (ArchanaKakani) [reply libpldm/pdr.c:79]: Yes, we are asserting in the case where we fail to add a record

File: libpldm/pdr.c (hunk 3 above)
Comment (ArchanaKakani) [reply libpldm/pdr.c:1723]: Done

**Review by jaypadath — APPROVED** [review]

**Review by Pavithrab7 — APPROVED** [review]

**Review by nmuruli — APPROVED** [review]

**Pull request merged**